addons/website_blog/static/src/scss/website_blog.scss

```diff
@@ -52,6 +52,19 @@ $o-wblog-loader-size: 50px;
         }
     }
 
+    > nav > .container {
+        > ul.navbar-nav {
+            flex-wrap: wrap;
+            flex-shrink: 2;
+        }
+        // If the navbar contains at least 10 items (random choice), the search
+        // input is placed at the top to avoid being centered in a multi-line
+        // navbar, which wouldn’t look good.
+        &:has(> ul.navbar-nav .nav-item:nth-child(10)) .o_searchbar_form {
+            align-self: start;
+        }
+    }
+
```

Comment:
Note for the reviewer: The CSS has been written to work in later versions as well.